src/Entity/Security/LocalAccount.php

```diff
@@ -329,9 +329,10 @@ public function setPasswordRequestSalt(?string $salt = null): LocalAccount
 
     public function getCanonical(): string
     {
+        assert(is_string($this->getId()));
         $pseudo = sprintf('pseudonymized (%s...)', substr($this->getId(), 0, 8));
 
-        return $this->getName() ?: $this->getEmail() ?: $pseudo;
+        return $this->getName() ?? $this->getEmail() ?? $pseudo;
```

Comment:
A shorthand ternary is correct here instead of a null coalescing operator, as an empty string is cause for changing the value. Please revert this change.

Reply:
Stan didn't like the short ternary, I'll just rewrite it as a nice `if, elseif, else`